### PR TITLE
Retrieve output data from device

### PR DESCRIPTION
**Pull request by mariecwhite**
Also temporarily fixes the `jax.DeviceArray` failure until https://github.com/huggingface/transformers/pull/24875 makes its way into a release.

### CODE REVIEW

File: .github/workflows/run_comparative_benchmark.yml
Comment (mariecwhite) [inline .github/workflows/run_comparative_benchmark.yml:16]: Remove before merge

File: comparative_benchmark/jax_xla/benchmark_all.sh
Comment (pzread) [inline comparative_benchmark/jax_xla/benchmark_all.sh:87]: How often does the transfomers library release? I'm trying to balance between a temporary patch and wait for a little bit long.

Alternatively, we can install the transformers from the main branch (maybe pinned at a commit). They seem to give out an official instruction to do that (https://huggingface.co/docs/transformers/installation#install-from-source).

Comment (mariecwhite) [reply comparative_benchmark/jax_xla/benchmark_all.sh:87]: It looks like they release every month and the last one happened ~3 weeks ago. We should expect one in the next week. Happy to wait or merge this and remove once it's released.

File: comparative_benchmark/jax_xla/run_benchmarks.py
Comment (mariecwhite) [reply comparative_benchmark/jax_xla/run_benchmarks.py:128]: GPU benchmarks were completing within 10 mins because almost all JAX workloads were failing. This is the first time it's run all benchmarks successfully so we do need to tune the timeout. The output data transfer latency does add to the total latency but generally not as high as the device latency.

File: comparative_benchmark/jax_xla/benchmark_all.sh
Comment (pzread) [reply comparative_benchmark/jax_xla/benchmark_all.sh:87]: Yeah, I prefer to wait for the release assuming it will come out next week. In the meantime we can tune the timeout problem on this PR

Comment (pzread) [reply comparative_benchmark/jax_xla/benchmark_all.sh:87]: Seems like they haven't released a new version yet. Feel free to file an issue and merge this

**Review by pzread — APPROVED** [review]

File: comparative_benchmark/jax_xla/benchmark_all.sh
Comment (mariecwhite) [reply comparative_benchmark/jax_xla/benchmark_all.sh:87]: Sounds good. I'll keep an eye out for a new HuggingFace release.

**Pull request merged**
